lib/vnet/app_resolver.go

```diff
@@ -108,19 +118,17 @@ func withClock(clock clockwork.Clock) tcpAppResolverOption {
 	}
 }
 
-// ResolveTCPHandler resolves a fully-qualified domain name (FQDN) to a TCPHandler for handling TCP
-// connections to a Teleport TCP app. It returns the TCPHandler if a matching app is found, else it will
-// return with match == false. Errors must only be returned for truly unexpected errors that should cause a
-// DNS request to fail.
-func (r *TCPAppResolver) ResolveTCPHandler(ctx context.Context, fqdn string) (handler TCPHandler, match bool, err error) {
+// ResolveTCPHandler resolves a fully-qualified domain name to a TCPHandlerSpec for a Teleport TCP app that should
+// be used to handle all future TCP connections to [fqdn].
+func (r *TCPAppResolver) ResolveTCPHandler(ctx context.Context, fqdn string) (*TCPHandlerSpec, error) {
 	profileNames, err := r.appProvider.ListProfiles()
 	if err != nil {
-		return nil, false, trace.Wrap(err, "listing profiles")
+		return nil, trace.Wrap(err, "listing profiles")
 	}
 	for _, profileName := range profileNames {
 		if fqdn == fullyQualify(profileName) {
 			// This is a query for the proxy address, which we'll never want to handle.
-			return nil, false, nil
+			return nil, ErrNoTCPHandler
```

Comment:
I specifically want to avoid `trace.Wrap` since this isn't an unexpected error and `trace.Wrap` would unnecessarily be collecting a full stack trace for every DNS lookup we're not handling, included this in the comment on ResolveTCPHandler in the TCPHandlerResolver interface

Reply:
added some more comments on this